src/bun.js/node/fs_events.zig

```diff
@@ -352,27 +358,46 @@ pub const FSEventsLoop = struct {
         var loop = bun.cast(*FSEventsLoop, info);
         const event_flags = bun.cast([*]FSEventStreamEventFlags, eventFlags);
 
+        var buffer: [bun.MAX_PATH_BYTES + 1]u8 = undefined;
+
         for (loop.watchers.slice()) |watcher| {
             if (watcher) |handle| {
+                // get resolved path once
+                if (handle.symlink_path == null) {
+                    handle.symlink_path = symlinkPath(handle.path, &buffer);
+                    if (handle.symlink_path) |path| {
+                        handle.symlink_path = bun.default_allocator.dupe(u8, path) catch |err| {
+                            handle.emit(@errorName(err), false, .@"error");
+                            handle.flush();
+                            continue;
+                        };
+                    } else {
+                        handle.symlink_path = "";
```

Comment:
did you mean to make symlink_path both nullable and empty? can we just make it empty and treat that as null?

Reply:
i meant to use null to check if I already tried to read the symlink (with can fail if is not a sym link) if is not this turns empty and I stop trying, but I will remove this logic FSEvents will already have the resolved path at this point.